wandb/agents/pyagent.py

```diff
@@ -27,6 +27,9 @@
 def _terminate_thread(thread):
     if not thread.is_alive():
         return
+    if hasattr(thread, "_terminated"):
+        return
+    thread._terminated = True
```

Comment:
So this is to catch the time between stopping a thread and it actually finishing execution?

Reply:
Yes.